src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionFactory.cs

```diff
@@ -0,0 +1,59 @@
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets
+{
+    /// <summary>
+    /// Defines a class for creating socket connections based on the specified endpoint.
+    /// </summary>
+    public class SocketConnectionFactory : IConnectionFactory
+    {
+        private readonly ILogger _logger;
+        private readonly SocketClientOptions _options;
+
+        /// <summary>
+        /// Creates the <see cref="SocketConnectionFactory"/>.
+        /// </summary>
+        /// <param name="options">The options for this transport</param>
+        /// <param name="loggerFactory">The logger factory</param>
+        public SocketConnectionFactory(IOptions<SocketClientOptions> options, ILoggerFactory loggerFactory)
```

Comment:
Do we want a client API to take IOptions? Do we expect this normally to be retrieved from a DI container?

Reply:
Tough question but a good one. The only client API we have now is DI enabled.

Reply:
You know how we solve this? We leave it DI friendly and then add static factory methods for the non DI case.

`var factory = SocketConnectionFactory.Create();`